src/boosting/gbdt_model_text.cpp

```diff
@@ -39,6 +39,21 @@ std::string GBDT::DumpModel(int start_iteration, int num_iteration) const {
   str_buf << "\"monotone_constraints\":["
           << Common::Join(monotone_constraints_, ",") << "]," << '\n';
 
+  str_buf << "\"feature_infos\":" << "{";
+  auto feature_infos_json_objs = train_data_->feature_infos<true>();
+  bool first_obj = true;
+  for (size_t i = 0; i < feature_infos_json_objs.size(); ++i) {
```

Comment:
it seems this loop causes the fails, @StrikerRUS could you check it ?

Reply:
I commented the loop and discovered that the following line causes segfaults:
```
auto feature_infos_json_objs = train_data_->feature_infos<true>();
```

Reply:
I think we cannot dynamically generate the `feature_infos_json_objs`, as the model could be loaded from file/string. In this case, both `used_feature_map_` and 'feature_bin_mappers_` are lost and cause the error.

Reply:
@StrikerRUS I propose a new change, you can check it.

Reply:
I see... Now `feature_infos` field is not actually a JSON representation. I think we can get back to my initial implementation then, which parses those strings and constructs valid JSONs.

Reply:
Yeah, it would be better

Reply:
Done in the latest commit.